### DISCUSSION TITLE
Container drill down for nested connections

### DISCUSSION THREAD

**Discussion by littleredshack** [jgraph/drawio#4190] [Ideas]
I work in large scale architectures for the London Stock Exchange. I am very keen to share some work I have been doing around architectures as code and to talk about these as potential features in draw.io. I love the product - feels slick and smooth to work with. Great job. I have a background in open source so hope we can start a discussion. 
Glen Stokes

**Comment by davidjgraph** [comment]
You just started one :) , maybe you could post the specifics?

**Reply by littleredshack** [reply]
I have a working prototype so it is easier to show you. 

An architecture in essence is just containers and connections. But when you have very large architectures you need a way to summarise things up. Layers do not work because everything goes up and down at the same time. 

I had this feature many many years ago in NetViz which was a large scale network diagramming tool.

The idea is that something can be a container.  Then if you put something inside that containers it is contained as a child. It moves around with the parent container.  And then you can have containers within containers endlessly.

Secondly, the links are intelligent and they summarise up. Imagine two parent containers each with multiple children. The respective children have links between them,  In the diagram if you are looking at all the detail then you would then see multiple links . But, if you double click on the two parents separately, then the diagram summarises up to a single link between the two parents. So you have intelligent summarisation of links.

The story is that you would look at a high level diagram for example between system A and system B. You would see a single link between System A and System B indicating that they are have a connection. If you double click on one pf the parents, say System A, you would see the children and now you would see links from each child up to System B. Then if you double click on System B you get the drill down detail and see the children connected.

**Reply by littleredshack** [reply]
This also introduces some very fun diagramming theory that has never actually been solved when you try to connect in layers infiiitely.

**Reply by davidjgraph** [reply]
Yeah, "could draw.io be just like this other tool I used to have" gets the answer no, unfortunately. draw.io is draw.io with it's specific scope. This sound like a product in it's own right.
